packages/core/src/modules/vc/models/credential/W3cCredentialStatus.ts

```diff
@@ -1,23 +1,61 @@
-import { IsString } from 'class-validator'
+import { plainToInstance } from 'class-transformer'
+import { IsEnum, IsString, validateOrReject } from 'class-validator'
 
+import { AgentContext } from '../../../../agent/context'
+import { CredoError } from '../../../../error'
 import { IsUri } from '../../../../utils/validators'
 
+import { CredentialStatus, CredentialStatusPurpose, CredentialStatusType } from './W3cJsonCredentialStatus'
+
 export interface W3cCredentialStatusOptions {
   id: string
-  type: string
+  type: CredentialStatusType
+  statusPurpose: CredentialStatusPurpose
+  statusListIndex: string
+  statusListCredential: string
 }
 
 export class W3cCredentialStatus {
   public constructor(options: W3cCredentialStatusOptions) {
     if (options) {
       this.id = options.id
       this.type = options.type
+      this.statusPurpose = options.statusPurpose
+      this.statusListIndex = options.statusListIndex
+      this.statusListCredential = options.statusListCredential
     }
   }
 
   @IsUri()
+  @IsString()
   public id!: string
 
+  @IsEnum(['BitstringStatusListEntry'])
+  @IsString()
+  public type!: CredentialStatusType
+
+  @IsEnum(CredentialStatusPurpose)
+  @IsString()
+  public statusPurpose!: CredentialStatusPurpose
+
   @IsString()
-  public type!: string
+  public statusListIndex!: string
```

Comment:
Looking at the VC Data model specification, these properties are not required fro the `credentialStatus` property. Only `id` and `type` are. 

We should have a generic base class and allow multiple extension points. You can look at e.g. how we solved this with the DidDocument service property, where there's a base service class and multiple possible extensions that are matched based on the `type` property